jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainActivity.java

```diff
@@ -1,68 +1,160 @@
+
+    }
 
-            // Edit the text view (show written message).
-            TextView textView = (TextView)
-                    display_message_fragment.getView().findViewById(R.id.display_message);
-            textView.setText(message);
+    /** Called when the user clicks the show/hide button.
+     *
+     * @param showDisplayMessageFragment True if the fragment that shows the message should be
+     *                                   visible on screen. */
+    public void showHide(Boolean showDisplayMessageFragment) {
 
-        // If orientation is in portrait start new activity.
+        // If the displaying fragment has not been initialized return.
+        if (displayMessageFragment == null) {
+            return;
+        }
+
+        // Setup FragmentManager and FragmentTransaction.
+        if (displayMessageFragmentContainer == null) {
+            displayMessageFragmentContainer =
+                    findViewById(R.id.fragment_container_display_message);
+        }
+
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        FragmentTransaction ft = fragmentManager.beginTransaction();
+        ft.setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out);
+
+        // Show or hide the DisplayMessageFragment.
+        if (showDisplayMessageFragment) {
+            ft.show(displayMessageFragment).commitNow();
+
+            // Set size of displayMessageFragment layout container to fill half the screen.
+            changeFragmentLayout(displayMessageFragmentContainer,
+                    fragmentLayouts.get(SHOW_FRAGMENT_LAYOUT));
         } else {
-            // Create intent.
-            Intent intent = new Intent(this, DisplayMessageActivity.class);
+            ft.hide(displayMessageFragment).commitNow();
 
-            // Add (user) entered message to the created intent.
-            intent.putExtra(EXTRA_MESSAGE, message);
+            int displayOrientation = getResources().getConfiguration().orientation;
 
-            // Start an instance of the DisplayMessageActivity specified by the intent.
-            startActivity(intent);
+            // Set main fragment to fill entire screen.
+            if (displayOrientation == LANDSCAPE) {
+                changeFragmentLayout(displayMessageFragmentContainer,
+                        fragmentLayouts.get(LANDSCAPE_HIDE_FRAGMENT_LAYOUT));
+            } else {
+                changeFragmentLayout(displayMessageFragmentContainer,
+                        fragmentLayouts.get(PORTRAIT_HIDE_FRAGMENT_LAYOUT));
+            }
         }
 
+    }
+
+    /** Helper function to change the layout of the fragment container.
+     *
+     * @param viewGroup The view that receives a layout change.
+     * @param newLayoutParams The new layout parameters that will be added to the given view. */
+    private void changeFragmentLayout(View viewGroup, ViewGroup.LayoutParams newLayoutParams) {
```

Comment:
I don't think we need this method anymore. What we want now is to set the layout params from the mapping. So we should just call:
```java
displayMessageFragmentContainer.setLayoutParams(fragmentLayouts.get(PORTRAIT_HIDE_FRAGMENT_LAYOUT));
```